### DISCUSSION TITLE
Mac mini 2018 kernel panic

### DISCUSSION THREAD

**Discussion by petercastles** [MonitorControl/MonitorControl#552] [Q&A] [answered]
Hi, I own a 2018 Mac mini, and I'm aware that I can't use hdmi for output since it doesn't support DDC/CI. 
So I bought a usb C - hdmi adapter (Anker).

Everything works fine until I launch MonitorControl: Mac immediately reboots and shows a kernel panic screen.  

Any advice? Thank you!

Edit: Lunar works just fine

**Comment by waydabber** [comment, marked as answer]
Hi,

one major difference might be that MonitorControl attempts to read your display's settings on startup while Lunar does not. This might cause issues with the dongle in question has issues with DDC communication. I advise the following:

- If you are on v3.0.0 RC2, press SHIFT during application startup. This will reset default settings and disable DDC read. If you can control the display this way, then enable 'Advanced settings' at the bottom of the General tab in Preferences and then set `DDC read polling mode` to `None` - this will permanently disable DDC read.
- You can also try v3.0.0 RC1 and v2.1.0 to see if you have the kernel panic with those versions - this will eliminate the possibility that changes in V3.0.0 RC2 causes the issue.
- If you can't fix the issue at all, then you can still disable `Use hardware DDC control` under Displays (you can get there by enabling Safe Mode in RC2).

Btw: Does Lunar work in Software mode or in Hardware mode?

Hope this helps! Please let me know the results!

**Reply by petercastles** [reply]
Thank you for your fast reply,

Pressing shift and then setting DDC read polling mode to None seems to fix the issue! 
If I then set that to normal again, still no problem until I relaunch the app -> kernel panic. 

Do you believe is an app problem or a dongle one? 
If I plan to always use keyboard to adjust brightness/volume (and I don't have a brightness sensor) I shouldn't have problem with polling on None, right?

PS: Lunar works in hw mode too.

**Reply by waydabber** [reply]
It seems to be a dongle issue. DDC communication is inherently a bit risky, macOS does not support it natively and private APIs are used. If hardware is not fully compliant then problems will arise.

You can set polling to `None` safely, this only disables reading actual display brightness/volume values but not affect changing them. If the app is prohibited from reading from the display upon startup then it expects that the last saved value to be correct (that you did not change the brightness manually). If you did change the values then the initial slider position will not be correct but upon first change things will be in sync again. In day to day use this does not matter, reading is just a convenience feature which happens only during app start. :)
